src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryUntypedObject.cs

```diff
@@ -200,6 +202,11 @@ public static FormulaValue DateTimeValue_UO(IRContext irContext, UntypedObjectVa
 
                 if (IsValidDateTimeUO(s) && DateTime.TryParse(s, CultureInfo.InvariantCulture, DateTimeStyles.None, out DateTime res))
                 {
+                    if (!res.IsValid(runner))
+                    {
+                        return CommonErrors.InvalidDateTimeError(irContext);
+                    }
+
```

Comment:
DateValue_UO and DateTimeValue move to the next valid date/time; DateTimeValue_UO returns an error; for consistency, this should also move to the next valid date/time